server/src/main/java/com/cloud/api/ApiDBUtils.java

```diff
@@ -1806,11 +1806,12 @@ public static List<DomainRouterJoinVO> newDomainRouterView(VirtualRouter vr) {
     }
 
     public static UserVmResponse newUserVmResponse(ResponseView view, String objectName, UserVmJoinVO userVm, EnumSet<VMDetails> details, Account caller) {
-        return s_userVmJoinDao.newUserVmResponse(view, objectName, userVm, details, null, caller);
+        return s_userVmJoinDao.newUserVmResponse(view, objectName, userVm, details, null, null, caller);
     }
 
-    public static UserVmResponse newUserVmResponse(ResponseView view, String objectName, UserVmJoinVO userVm, EnumSet<VMDetails> details, Boolean accumulateStats, Account caller) {
-        return s_userVmJoinDao.newUserVmResponse(view, objectName, userVm, details, accumulateStats, caller);
+    public static UserVmResponse newUserVmResponse(ResponseView view, String objectName, UserVmJoinVO userVm, EnumSet<VMDetails> details, Boolean accumulateStats,
```

Comment:
sonar is making a valid complaint about this use of `EnumSet<>` it is done in all other methods around here but we should not add to that. Can you please
```suggestion
    public static UserVmResponse newUserVmResponse(ResponseView view, String objectName, UserVmJoinVO userVm, Set<VMDetails> details, Boolean accumulateStats,
```
for your new method and all others? (obvioudsly this is a request not a blocking remarks on your code)

Reply:
Yeah, no problem :)